pkg/hubble/exporter/config.go

```diff
@@ -0,0 +1,237 @@
+		} else {
+			if _, ok := flowlogPaths[filePath]; ok {
+				errs = errors.Join(errs, fmt.Errorf("duplicated flowlog path %s", filePath))
+			}
+			flowlogPaths[filePath] = struct{}{}
+		}
+	}
+
+	return errs
+}
+
+var _ ExporterConfig = (*FlowLogConfig)(nil)
+
+// FlowLogConfig represents configuration of single dynamic exporter.
+type FlowLogConfig struct {
+	Name           string      `json:"name,omitempty" yaml:"name,omitempty"`
+	FilePath       string      `json:"filePath,omitempty" yaml:"filePath,omitempty"`
+	FieldMask      FieldMask   `json:"fieldMask,omitempty" yaml:"fieldMask,omitempty"`
+	IncludeFilters FlowFilters `json:"includeFilters,omitempty" yaml:"includeFilters,omitempty"`
+	ExcludeFilters FlowFilters `json:"excludeFilters,omitempty" yaml:"excludeFilters,omitempty"`
+	FileMaxSizeMB  int         `json:"fileMaxSizeMb,omitempty" yaml:"fileMaxSizeMb,omitempty"`
+	FileMaxBackups int         `json:"fileMaxBackups,omitempty" yaml:"fileMaxBackups,omitempty"`
+	FileCompress   bool        `json:"fileCompress,omitempty" yaml:"fileCompress,omitempty"`
+	End            *time.Time  `json:"end,omitempty" yaml:"end,omitempty"`
+}
+
+// Equal implements ExporterConfig.
+func (f *FlowLogConfig) Equal(other any) bool {
+	if other, ok := other.(*FlowLogConfig); ok {
+		return f.equals(other)
+	}
+	return false
+}
+
+// Equal implements ExporterConfig.
+func (f *FlowLogConfig) IsActive() bool {
+	return f.End == nil || f.End.After(time.Now())
+}
+
+func (f *FlowLogConfig) equals(other *FlowLogConfig) bool {
+	if f == nil && other == nil {
+		return true
+	}
+	if f == nil || other == nil {
+		return false
+	}
+
+	if f.FilePath != other.FilePath {
+		return false
+	}
+
+	if !f.FieldMask.equals(other.FieldMask) {
+		return false
+	}
+
+	if !f.IncludeFilters.equals(other.IncludeFilters) {
+		return false
+	}
+
+	if !f.ExcludeFilters.equals(other.ExcludeFilters) {
+		return false
+	}
+
+	if f.FileMaxSizeMB != other.FileMaxSizeMB {
+		return false
+	}
+
+	if f.FileMaxBackups != other.FileMaxBackups {
+		return false
+	}
+
+	if f.FileCompress != other.FileCompress {
+		return false
+	}
+
+	if f.End == nil && other.End != nil ||
+		f.End != nil && other.End == nil ||
+		f.End != nil && other.End != nil && !f.End.Equal(*other.End) {
+		return false
+	}
+
+	return true
+}
+
+// FlowFilters is a slice of filters to include or exclude flows.
+type FlowFilters []*flowpb.FlowFilter
+
+func (f FlowFilters) equals(other FlowFilters) bool {
+	aFiltersSet, bFiltersSet := make(map[string]bool), make(map[string]bool)
+
+	for _, filter := range f {
+		aFiltersSet[filter.String()] = true
+	}
+	for _, filter := range other {
+		bFiltersSet[filter.String()] = true
+	}
+	return reflect.DeepEqual(aFiltersSet, bFiltersSet)
+}
+
+// FieldMask is a slice of fields that are included in output.
+type FieldMask []string
+
+func (f FieldMask) equals(other FieldMask) bool {
+	slices.Sort(f)
+	slices.Sort(other)
+	return reflect.DeepEqual(f, other)
```

Comment:
Slightly worried about mutation in `equals`, though semantically it shouldn't have any impact. To err on the safe side I would favor a solution that keeps `f` and `other` const, e.g.
```go
    // slow, but n is small
    xs := make([]string, len(f))
    ys := make([]string, len(other))
    copy(xs, f)
    copy(ys, other)
    slices.Sort(xs)
    slices.Sort(ys)
    return reflect.DeepEqual(xs, ys)
```